lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -350,3 +352,185 @@ def _get(self, json_data: JSON, _) -> Response:
+                not be altered unless the error_count is 0.
+        """
+        raise NotImplementedError()
+
+    def _post(self, json_data: JSON, _) -> Response:
+        """
+        Perform the requested POST operation, and handle any exceptions.
+
+        NOTE: This is called by the ApiBase post() method through its dispatch
+        method, which provides parameter validation.
+
+        Args:
+            json_data: Type-normalized client JSON input
+            _: Original incoming Request object (not used)
+
+        Returns:
+            Response to return to client
+        """
+        klasname = self.__class__.__name__
+
+        try:
+            dataset = Dataset.attach(
+                controller=json_data["controller"], name=json_data["name"]
+            )
+        except DatasetNotFound as e:
+            abort(HTTPStatus.NOT_FOUND, message=str(e))
+
+        owner = User.query(id=dataset.owner_id)
+        if not owner:
+            self.logger.error(
+                "Dataset owner ID {} cannot be found in Users", dataset.owner_id
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="Dataset owner not found")
+
+        # For bulk Elasticsearch operations, we check authorization against the
+        # ownership of a designated dataset rather than having an explicit
+        # "user" JSON parameter. This will raise UnauthorizedAccess on failure.
+        try:
+            self._check_authorization(owner.username, json_data["access"])
+        except UnauthorizedAccess as e:
+            abort(HTTPStatus.FORBIDDEN, message=str(e))
+
+        # Build an Elasticsearch instance to manage the bulk update
+        elastic = elasticsearch.Elasticsearch([self.es_url])
+
+        try:
+            # Pass the bulk command generator to the helper
+            results = elasticsearch.helpers.streaming_bulk(
+                elastic,
+                self.generate_documents(json_data, dataset),
+                raise_on_exception=False,
+                raise_on_error=False,
+            )
+        except Exception as e:
+            self.logger.exception(
+                "{}: exception {} occurred during the Elasticsearch request",
+                klasname,
+                type(e).__name__,
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+
+        report = {}
+        count = 0
+        error_count = 0
+        for ok, response in results:
+            count += 1
+            u = response["update"]
+            status = "ok"
+            if "error" in u:
+                e = u["error"]
+                status = e["type"]
+                self.logger.debug(
+                    "{} ({}: {}) for id {} in index {}",
+                    u["status"],
+                    status,
+                    e["reason"],
+                    u["_id"],
+                    u["_index"],
+                )
```

Comment:
`report` is the client response; I don't want Elasticsearch details there. I carried through the ES "status" keyword to provide some loose categorization of errors, although I wasn't entirely sure about that. I definitely don't want to return document IDs and index names; but in case of failure, I *did* want them visible somewhere.

Although since I decided to avoid logging them at `info` level this is to some extent moot as we won't see it in a real server deployment.

Reply:
> `report` is the client response; I don't want Elasticsearch details there.

Good point.

> we won't see it in a real server deployment.

Perhaps not...but have some pity on the guy debugging this stuff!  😉